pulsar-client-cpp/lib/BatchMessageContainer.cc

```diff
@@ -117,7 +117,8 @@ SharedBuffer BatchMessageContainer::getBatchedPayload() {
 void BatchMessageContainer::clear() {
     LOG_DEBUG(*this << " BatchMessageContainer::clear() called");
     timer_->cancel();
-    averageBatchSize_ = (messagesContainerListPtr_->size() + (averageBatchSize_ * numberOfBatchesSent_))/++numberOfBatchesSent_;
+    averageBatchSize_ = (messagesContainerListPtr_->size() + (averageBatchSize_ * numberOfBatchesSent_))/(numberOfBatchesSent_ + 1);
```

Comment:
yes, the compiler doesn't give ordering guarantees for ++ if used in complex calculations
Try compiling in your mac m/c and you will see the warning.